state/reader.go

```diff
@@ -0,0 +1,41 @@
+package state
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/xh3b4sd/anna/common"
+)
+
+func (s *state) Read() error {
+	switch s.StateReader {
+	case common.FSStateReader:
+		err := s.ReadFile(common.DefaultStateFile)
+		if err != nil {
+			return maskAny(err)
+		}
+	case common.NoneStateReader:
+		// Do nothing.
+	default:
+		return maskAny(invalidStateReaderError)
+	}
+
+	return nil
+}
+
+func (s *state) ReadFile(filename string) error {
+	bytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return maskAny(err)
+	}
+
+	var newState state
+	err = json.Unmarshal(bytes, &newState)
```

Comment:
The state object references to a lot of interface implementations. The golang json package cannot know what specific type to use to unmarshal bytes into. We need to write the specific unmarshaling ourself.